src/ln/channelmanager.rs

```diff
@@ -1147,7 +1147,12 @@ impl ChannelManager {
 					if !add_htlc_msgs.is_empty() {
 						let (commitment_msg, monitor) = match forward_chan.send_commitment() {
 							Ok(res) => res,
-							Err(_e) => {
+							Err(e) => {
+								if let &Some(msgs::ErrorAction::DisconnectPeer{msg: Some(ref _err_msg)}) = &e.action {
+								} else if let &Some(msgs::ErrorAction::SendErrorMessage{msg: ref _err_msg}) = &e.action {
+								} else {
+									panic!("Stated return value requirements in send_commitment() were not met");
+								}
 								//TODO: Handle...this is bad!
```

Comment:
seems all kind of errors are covered by the last else, so TODO can disappear?

Reply:
Heh, the ifs/panic is just to give the fuzz tester something to attack. The actually *handling* still has to happen.